### PR TITLE
Export UI Images according to full in-game path

### PR DESCRIPTION
**Pull request by EladNLG**
See title.

### CODE REVIEW

File: Legion/src/Assets/material.cpp [hunk 1]
```diff
@@ -64,7 +64,15 @@ RMdlMaterial RpakLib::ExtractMaterial(const RpakLoadAsset& Asset, const string&
...
-	Result.MaterialName = IO::Path::GetFileNameWithoutExtension(Reader.ReadCString());
+	string MaterialPath = Reader.ReadCString();
+
+	Result.MaterialName = IO::Path::GetFileNameWithoutExtension(MaterialPath);
+
+	const size_t last_slash_idx = MaterialPath.LastIndexOf('\\');
+	if (std::string::npos != last_slash_idx)
+	{
+		Result.MaterialPath = MaterialPath.Substring(0, last_slash_idx);
+	}
```
Comment (r-ex) [inline Legion/src/Assets/material.cpp:75]: not sure why this is here considering this is unrelated to the PR and isn't even used

File: Legion/src/Assets/uimg.cpp [hunk 2]
```diff
@@ -115,8 +115,18 @@ void RpakLib::ExtractUIImageAtlas(const RpakLoadAsset& Asset, const string& Path
...
+			ImageName = string::Format("%s%s", img.Path.ToCString(), (const char*)ImageExtension);
 
-		tmp->Save(IO::Path::Combine(Path, ImageName), ImageSaveType);
+
+		const size_t last_slash_idx = ImageName.LastIndexOf('/');
+		string ImageDirectory = "";
+		if (std::string::npos != last_slash_idx)
+		{
+			ImageDirectory = ImageName.Substring(0, last_slash_idx);
+		}
```
Comment (r-ex) [inline Legion/src/Assets/uimg.cpp:126]: this is unnecessary - use [IO::Path::GetDirectoryName](https://github.com/r-ex/LegionPlus/blob/814acd4a791878efe0701a264b37eff867a3d26d/cppnet/cppkore/Path.h#L28)

**Comment by EladNLG** [comment]
Should probably be made into a setting before this gets merged.
